scripts.py

```diff
@@ -1381,22 +1402,26 @@ def load_collection(self, collection_name, data_source_name):
             )
 
             try:
-                mirror_integration = MirrorUploader.sitewide_integration(
-                    self._db
+                mirror_integration = MirrorUploader.integration_by_name(
+                    self._db, storage_name
                 )
             except CannotLoadConfiguration, e:
-                # There is no sitewide mirror configuration, or else
-                # there is more than one. Either way, we can't
-                # associate a mirror integration with the new collection.
+                # We can't associate any mirror integration with the new collection.
                 mirror_integration = None
 
             if mirror_integration:
                 collection.mirror_integration = mirror_integration
```

Comment:
I don't believe this is needed anymore and can be removed, but should the `mirror_integration_id` in the Collections model be removed since now there can be multiple integrations?

Reply:
Yes, we can and should remove `collections.mirror_integration_id`. We'll need a migration script that creates two ExternalIntegrationLinks for every collection with `mirror_integration_id` set -- one to say that book covers use the S3 integration and another one to say that books use the same S3 integration.